mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,433 @@
+
+
+class CsvEvaluationArtifact(EvaluationArtifact):
+    def save(self, output_artifact_path):
+        self._content.to_csv(output_artifact_path, index=False)
+
+    def _load_content_from_file(self, local_artifact_path):
+        self._content = pd.read_csv(local_artifact_path)
+        return self._content
+
+
+_DEFAULT_SAMPLE_ROWS_FOR_SHAP = 2000
+
+
+_shap_initialized = False
+
+
+class DefaultEvaluator(ModelEvaluator):
+    def can_evaluate(self, model_type, evaluator_config=None, **kwargs):
+        return model_type in ["classifier", "regressor"]
+
+    @staticmethod
+    def _gen_log_key(key, dataset_name):
+        return f'{key}_on_data_{dataset_name}'
+
+    def _log_metrics(self, run_id, metrics, dataset_name):
+        """
+        Helper method to log metrics into specified run.
+        """
+        client = mlflow.tracking.MlflowClient()
+        timestamp = int(time.time() * 1000)
+        client.log_batch(
+            run_id,
+            metrics=[
+                Metric(key=DefaultEvaluator._gen_log_key(key, dataset_name),
+                       value=value, timestamp=timestamp, step=0)
+                for key, value in metrics.items()
+            ],
+        )
+
+    def _log_image_artifact(
+        self, artifacts, temp_dir, do_plot, run_id, artifact_name, dataset_name,
+    ):
+        import matplotlib.pyplot as pyplot
+        client = mlflow.tracking.MlflowClient()
+        pyplot.clf()
+        do_plot()
+        artifact_file_name = DefaultEvaluator._gen_log_key(artifact_name, dataset_name) + '.png'
+        artifact_file_local_path = temp_dir.path(artifact_file_name)
+        pyplot.savefig(artifact_file_local_path)
+        client.log_artifact(run_id, artifact_file_local_path)
+        artifact = ImageEvaluationArtifact(uri=get_artifact_uri(run_id, artifact_file_name))
+        artifact.load(artifact_file_local_path)
+        artifacts[artifact_name] = artifact
+        pyplot.close(pyplot.gcf())
+
+    def _log_pandas_df_artifact(
+        self, artifacts, temp_dir, pandas_df, run_id, artifact_name, dataset_name, model
+    ):
+        client = mlflow.tracking.MlflowClient()
+        artifact_file_name = DefaultEvaluator._gen_log_key(artifact_name, dataset_name) + '.csv'
+        artifact_file_local_path = temp_dir.path(artifact_file_name)
+        pandas_df.to_csv(artifact_file_local_path, index=False)
+        client.log_artifact(run_id, artifact_file_local_path)
+        artifact = CsvEvaluationArtifact(
+            uri=get_artifact_uri(run_id, artifact_file_name),
+            content=pandas_df,
+        )
+        artifact.load(artifact_file_local_path)
+        artifacts[artifact_name] = artifact
+
+    def _extract_raw_model_and_predict_fn(self, model):
+        model_loader_module = model.metadata.flavors['python_function']["loader_module"]
+        predict_fn = model.predict
+        predict_proba_fn = None
+
+        try:
+            if model_loader_module == 'mlflow.sklearn':
```

Comment:
```suggestion
            if model_loader_module == mlflow.sklearn.__name__:
```

Can we de-hardcode this?

Reply:
The loader module name should remain stable across MLflow versions; in that sense, hardcoding is safer.

Reply:
Makes sense!